src/compiler/scala/tools/nsc/backend/jvm/opt/CallGraph.scala

```diff
@@ -295,6 +295,7 @@ class CallGraph[BT <: BTypes](val btypes: BT) {
           // TODO: type analysis can render more calls statically resolved. Example:
           //   new A.f  // can be inlined, the receiver type is known to be exactly A.
           val isStaticallyResolved: Boolean = {
+            isNonVirtualCall(call) || // SD-86: super calls (invokespecial) can be inlined
```

Comment:
Noting that although this will include constructors, they are explicitly excluded below with `!BytecodeUtils.isConstructor(calleeMethodNode)`